src/foundry/client/canvas/interaction/mouse-handler.d.mts

```diff
@@ -1,78 +1,99 @@
+ *   - action: dragLeftStart
+ *   - action: dragRightStart
+ *   - action: dragLeftMove
+ *   - action: dragRightMove
+ * - _handlePointerUp
+ *   - action: dragLeftDrop
+ *   - action: dragRightDrop
+ * - _handleDragCancel
+ *   - action: dragLeftCancel
+ *   - action: dragRightCancel
  */
 declare class MouseInteractionManager<ObjectFor extends PIXI.Container = PIXI.Container> {
   /**
-   * @param permissions - (default: `{}`)
-   * @param callbacks   - (default: `{}`)
-   * @param options     - (default: `{}`)
-   * @remarks Foundry does not provide type information for the constructor, everything here is inferred from usage
+   * @param object       - The Canvas object (e.g., a Token, Tile, or Drawing) to which mouse events should be bound.
+   * @param layer        - The Canvas Layer that contains the object.
+   * @param permissions  - An object of permission checks, keyed by action name, which return a boolean or invoke a function for whether the action is allowed. (default: `{}`)
+   * @param callbacks    - An object of callback functions, keyed by action name, which will be executed during the event workflow (e.g., `hoverIn`, `clickLeft`). (default: `{}`)
+   * @param options      - Additional options that configure interaction behavior. (default: `{}`)
+   *
+   * @remarks Foundry types `object` as {@linkcode PIXI.DisplayObject}, but only ever passes a {@linkcode placeables.PlaceableObject | PlaceableObject}
+   * (in {@linkcode placeables.PlaceableObject._createInteractionManager | #_createInteractionManager}) or the {@linkcode Canvas.stage | canvas.stage}
+   * (in `Canvas##addListeners`), so it's typed as {@linkcode PIXI.Container} here.
+   *
+   * `layer` is described as if it's a {@linkcode foundry.canvas.layers.CanvasLayer | CanvasLayer}, but Foundry always passes the {@linkcode Canvas.stage | canvas.stage}
    */
   constructor(
     object: ObjectFor,
     layer: PIXI.Container,
-    permissions?: MouseInteractionManager.Permissions,
-    callbacks?: MouseInteractionManager.Callbacks,
+    permissions?: MouseInteractionManager.Permissions<ObjectFor>,
+    callbacks?: MouseInteractionManager.Callbacks<ObjectFor>,
     options?: MouseInteractionManager.Options,
   );
 
+  /**
+   * The Canvas object (e.g., a Token, Tile, or Drawing) to which mouse events should be bound.
+   * @remarks See remarks on {@linkcode MouseInteractionManager | MouseInteractionManager#constructor}
+   */
   object: ObjectFor;
 
+  /**
+   * The Canvas Layer that contains the object.
+   * @remarks Despite the above description, in core usage this is always the {@linkcode Canvas.stage | canvas.stage}
+   */
   layer: PIXI.Container;
 
   /**
+   * An object of permission checks, keyed by action name, which return a boolean or invoke a function for whether the action is allowed.
    * @defaultValue `{}`
    */
+  // TODO: This should be MouseInteractionManager.Permissions<ObjectFor> but that causes circularities in all placeables
```

Comment:
I'll take a look but I might split it into its own ticket since I have my hands full with some other circularities this weekend.